src/Analysis/Engine/Impl/PythonAnalyzer.cs

```diff
@@ -751,43 +751,50 @@ private static IMemberResult[] MemberDictToMemberResult(Dictionary<string, List<
             }
 
             var attrType = attr.GetType();
-            if (attr is IPythonType pt) {
-                return GetBuiltinType(pt);
-            } else if (attr is IPythonFunction pf) {
-                return GetCached(attr, () => new BuiltinFunctionInfo(pf, this)) ?? _noneInst;
-            } else if (attr is IPythonMethodDescriptor md) {
-                return GetCached(attr, () => {
-                    if (md.IsBound) {
-                        return new BuiltinFunctionInfo(md.Function, this);
-                    } else {
-                        return new BuiltinMethodInfo(md, this);
-                    }
-                }) ?? _noneInst;
-            } else if (attr is IPythonBoundFunction pbf) {
-                return GetCached(attr, () => new BoundBuiltinMethodInfo(pbf, this)) ?? _noneInst;
-            } else if (attr is IBuiltinProperty bp) {
+            if (attr is IPythonFunction pf) {
```

Comment:
Looks like a good candidate for `switch` statement, but up to you.

Reply:
All this will go away when DDG is replaced by AST analysis.